app/models/derived_variables/sales_log_variables.rb

```diff
@@ -16,5 +16,77 @@ def set_derived_fields!
       self.mscharge = 0
     end
     self.pcode1, self.pcode2 = postcode_full.split(" ") if postcode_full.present?
+    self.totchild = total_child
+    self.totadult = total_adult + total_elder
+    self.hhmemb = totchild + totadult
+    self.hhtype = household_type
+  end
+
+private
+
+  def total_elder
+    ages = [age1, age2, age3, age4, age5, age6]
+    ages.count { |age| age.present? && age >= 60 }
+  end
+
+  def total_child
+    (2..6).count do |i|
+      age = public_send("age#{i}")
+      relat = public_send("relat#{i}")
+      age.present? && ((age < 20 && %w[C].include?(relat)))
```

Comment:
@kosiakkatrina in the ticket Jemma wrote the following:

> For tenants that aren’t the lead tenant, we define adult as 18 or more, though if the person has RELAT = C and is less than 20, we define them as a child. 

Is the code above not missing the logic for people who are 16 or 17 but don't have RELAT = C (and who aren't the lead tenant of course).

Reply:
Sorry, merged this without seeing the comment, but will look at this now

Reply:
https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1195